weather_report/main.py

```diff
@@ -0,0 +1,172 @@
+                    highest_humidity_idx=idx
+        highest_temp_row=data[highest_temp_idx]
+        lowest_temp_row=data[lowest_temp_idx]
+        highest_humidity_row=data[highest_humidity_idx]
+        print("Highest:",str(highest_temp_row['Max TemperatureC'])+
+        'C on ',highest_temp_row['PKT'].strftime("%B %d"))
+        print("Lowest:",str(lowest_temp_row['Min TemperatureC'])+
+        'C on ',lowest_temp_row['PKT'].strftime("%B %d"))
+        print("Humidity:",str(highest_humidity_row['Max Humidity'])+
+        '% on ',highest_humidity_row['PKT'].strftime("%B %d"))
+
+@year_month_validation_decorator
+def get_month_avg_report(data,r_year,r_month)->None:
+    """
+    For a given month it displaysr the average highest temperature,
+    average lowest temperature, average mean humidity.
+    """
+    sum_highest_temp=0.0
+    sum_lowest_temp=0.0
+    sum_mean_humidity=0.0
+    month_length=0
+    for row in data:
+        if row['PKT'].year== r_year and row['PKT'].month==r_month:
+            month_length+=1
+            sum_highest_temp+=row['Max TemperatureC']
+            sum_lowest_temp+=row['Min TemperatureC']
+            sum_mean_humidity+=row['Mean Humidity']
+    avg_highest_temp=round(sum_highest_temp/month_length,1)
+    avg_lowest_temp=round(sum_lowest_temp/month_length,1)
+    avg_mean_humidity=round(sum_mean_humidity/month_length,1)
+    print(f"Highest Average: {avg_highest_temp}C")
+    print(f"Lowest Average: {avg_lowest_temp}C")
+    print(f"Average Mean Humidity: {avg_mean_humidity}%")
+
+@year_month_validation_decorator
+def get_month_temp_with_bars(data,r_year,r_month) -> None:
+    """
+    For a given month it draws two horizontal bar charts on the console for the highest
+    and lowest temperature on each day. Highest in red and lowest in blue.
+    """
+    # ANSI escape codes for text colors
+    RESET = '\033[0m' 
+    RED = '\033[91m'
+    BLUE = '\033[94m'
+    for row in data:
+        if row['PKT'].year== r_year and row['PKT'].month==r_month:
+            min_temp=int(row['Min TemperatureC'])
+            max_temp=int(row['Max TemperatureC'])
+            if max_temp:
+                high_temp_str=RED
+                max_temp=int(row['Max TemperatureC'])
+                for _ in range(max_temp):
+                    high_temp_str+='+'
+                high_temp_str+=RESET
+            if min_temp:
+                low_temp_str=BLUE
+                for _ in range(min_temp):
+                    low_temp_str+='+'
+                low_temp_str+=RESET
+                temp_str=row['PKT'].strftime("%d")+" "
+                temp_str+=low_temp_str+high_temp_str+" "
+                temp_str+= str(min_temp)+"C - "+ str(max_temp)+"C"
+                print(temp_str)
+
+if __name__=="__main__":
+    parser=argparse.ArgumentParser(description="Weather Report file")
+    parser.add_argument('-e', '--year',
+    help='Takes year as input for year report')
```

Comment:
If you specify `type=int` here, then you'll avoid typecasting this value to int everywhere in your code.